INFO 2024-03-23 23:58:03,989 autoreload 12300 15724 Watching for file changes with StatReloader
INFO 2024-03-23 23:58:06,527 basehttp 12300 14772 "GET / HTTP/1.1" 200 13
INFO 2024-03-23 23:58:13,279 basehttp 12300 14772 "GET / HTTP/1.1" 200 13
INFO 2024-03-23 23:58:20,989 basehttp 12300 14772 "GET /cub/ HTTP/1.1" 200 1
INFO 2024-03-27 22:13:01,056 autoreload 18028 16176 Watching for file changes with StatReloader
INFO 2024-03-27 22:13:02,660 basehttp 18028 12456 "GET / HTTP/1.1" 200 13
WARNING 2024-03-27 22:13:02,926 log 18028 12456 Not Found: /favicon.ico
WARNING 2024-03-27 22:13:02,927 basehttp 18028 12456 "GET /favicon.ico HTTP/1.1" 404 2951
INFO 2024-03-27 22:13:09,578 basehttp 18028 12456 "GET /about HTTP/1.1" 301 0
INFO 2024-03-27 22:13:09,582 basehttp 18028 10420 "GET /about/ HTTP/1.1" 200 8
INFO 2024-03-27 22:13:34,449 basehttp 18028 10420 "GET /about/ HTTP/1.1" 200 8
INFO 2024-03-27 22:13:42,049 basehttp 18028 10420 "GET /heads HTTP/1.1" 301 0
INFO 2024-03-27 22:13:42,051 basehttp 18028 16100 "GET /heads/ HTTP/1.1" 200 10
INFO 2024-03-27 22:13:45,369 basehttp 18028 16100 "GET /heads/ HTTP/1.1" 200 8
INFO 2024-03-27 22:13:48,939 basehttp 18028 16100 "GET /cub HTTP/1.1" 301 0
INFO 2024-03-27 22:13:48,941 basehttp 18028 5844 "GET /cub/ HTTP/1.1" 200 1
INFO 2024-03-27 22:13:54,263 basehttp 18028 5844 "GET /cub/ HTTP/1.1" 200 1
INFO 2024-03-27 22:13:58,300 basehttp 18028 5844 "GET /numbers HTTP/1.1" 301 0
INFO 2024-03-27 22:13:58,303 basehttp 18028 10040 "GET /numbers/ HTTP/1.1" 200 3
INFO 2024-03-27 22:26:32,944 autoreload 18028 16176 D:\GeekBrains\Django\Seminars\myproject\myapp\models.py changed, reloading.
INFO 2024-03-27 22:35:42,215 autoreload 15136 5000 Watching for file changes with StatReloader
INFO 2024-03-27 22:36:11,865 autoreload 15136 5000 D:\GeekBrains\Django\Seminars\myproject\myapp\views.py changed, reloading.
INFO 2024-03-27 22:36:12,265 autoreload 16012 13780 Watching for file changes with StatReloader
INFO 2024-03-27 22:36:15,785 autoreload 11788 4672 Watching for file changes with StatReloader
INFO 2024-03-27 22:36:19,314 basehttp 11788 3908 "GET / HTTP/1.1" 200 13
ERROR 2024-03-27 22:36:25,003 log 11788 3908 Internal Server Error: /heads/
Traceback (most recent call last):
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\models\fields\__init__.py", line 2117, in get_prep_value
    return int(value)
           ^^^^^^^^^^
ValueError: invalid literal for int() with base 10: 'Решка'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\Seminars\myproject\myapp\views.py", line 17, in heads_or_tails
    choice_coin.save()
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\models\base.py", line 822, in save
    self.save_base(
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\models\base.py", line 909, in save_base
    updated = self._save_table(
              ^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\models\base.py", line 1036, in _save_table
    updated = self._do_update(
              ^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\models\base.py", line 1080, in _do_update
    filtered = base_qs.filter(pk=pk_val)
               ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\models\query.py", line 1476, in filter
    return self._filter_or_exclude(False, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\models\query.py", line 1494, in _filter_or_exclude
    clone._filter_or_exclude_inplace(negate, args, kwargs)
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\models\query.py", line 1501, in _filter_or_exclude_inplace
    self._query.add_q(Q(*args, **kwargs))
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\models\sql\query.py", line 1613, in add_q
    clause, _ = self._add_q(q_object, self.used_aliases)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\models\sql\query.py", line 1645, in _add_q
    child_clause, needed_inner = self.build_filter(
                                 ^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\models\sql\query.py", line 1559, in build_filter
    condition = self.build_lookup(lookups, col, value)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\models\sql\query.py", line 1389, in build_lookup
    lookup = lookup_class(lhs, rhs)
             ^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\models\lookups.py", line 30, in __init__
    self.rhs = self.get_prep_lookup()
               ^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\models\lookups.py", line 364, in get_prep_lookup
    return super().get_prep_lookup()
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\models\lookups.py", line 88, in get_prep_lookup
    return self.lhs.output_field.get_prep_value(self.rhs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\models\fields\__init__.py", line 2119, in get_prep_value
    raise e.__class__(
ValueError: Field 'id' expected a number but got 'Решка'.
ERROR 2024-03-27 22:36:25,008 basehttp 11788 3908 "GET /heads/ HTTP/1.1" 500 143400
INFO 2024-03-27 22:37:27,225 autoreload 11788 4672 D:\GeekBrains\Django\Seminars\myproject\myapp\views.py changed, reloading.
INFO 2024-03-27 22:37:30,566 autoreload 16264 5760 Watching for file changes with StatReloader
INFO 2024-03-27 22:37:32,108 basehttp 16264 15764 "GET / HTTP/1.1" 200 13
INFO 2024-03-27 22:37:37,167 basehttp 16264 15764 "GET /heads/ HTTP/1.1" 200 116
INFO 2024-03-27 22:38:22,628 autoreload 16264 5760 D:\GeekBrains\Django\Seminars\myproject\myapp\migrations\0001_initial.py changed, reloading.
INFO 2024-03-27 22:38:24,948 autoreload 14292 11884 Watching for file changes with StatReloader
INFO 2024-03-27 22:38:26,193 basehttp 14292 17912 "GET / HTTP/1.1" 200 13
INFO 2024-03-27 22:38:32,167 basehttp 14292 17912 "GET /heads/ HTTP/1.1" 200 116
INFO 2024-03-27 22:57:45,973 autoreload 14292 11884 D:\GeekBrains\Django\Seminars\myproject\myapp\views.py changed, reloading.
INFO 2024-03-27 22:57:46,395 autoreload 328 17404 Watching for file changes with StatReloader
INFO 2024-03-27 23:01:35,934 autoreload 328 17404 D:\GeekBrains\Django\Seminars\myproject\myapp\views.py changed, reloading.
INFO 2024-03-27 23:01:36,379 autoreload 18088 15968 Watching for file changes with StatReloader
INFO 2024-03-27 23:01:44,717 autoreload 14292 12104 Watching for file changes with StatReloader
INFO 2024-03-27 23:01:45,857 basehttp 14292 8528 "GET / HTTP/1.1" 200 13
INFO 2024-03-27 23:01:55,989 basehttp 14292 8528 "GET /coin_values HTTP/1.1" 301 0
INFO 2024-03-27 23:01:55,995 basehttp 14292 11612 "GET /coin_values/ HTTP/1.1" 200 5
INFO 2024-03-27 23:08:03,400 autoreload 14292 12104 D:\GeekBrains\Django\Seminars\myproject\myapp\views.py changed, reloading.
INFO 2024-03-27 23:08:03,810 autoreload 16476 1880 Watching for file changes with StatReloader
INFO 2024-03-27 23:08:06,531 autoreload 14636 10116 Watching for file changes with StatReloader
INFO 2024-03-27 23:08:07,685 basehttp 14636 15564 "GET / HTTP/1.1" 200 13
ERROR 2024-03-27 23:08:14,155 log 14636 15564 Internal Server Error: /coin_values/
Traceback (most recent call last):
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\Seminars\myproject\myapp\views.py", line 22, in coin_values
    return HttpResponse(val for val in value)
                       ^^^^^^^^^^^^^^^^^^^^^^
TypeError: 'NoneType' object is not iterable
ERROR 2024-03-27 23:08:14,157 basehttp 14636 15564 "GET /coin_values/ HTTP/1.1" 500 65111
INFO 2024-03-27 23:10:30,119 autoreload 14636 10116 D:\GeekBrains\Django\Seminars\myproject\myapp\views.py changed, reloading.
INFO 2024-03-27 23:10:30,530 autoreload 10328 11932 Watching for file changes with StatReloader
INFO 2024-03-27 23:10:33,240 autoreload 11800 15576 Watching for file changes with StatReloader
INFO 2024-03-27 23:10:34,461 basehttp 11800 11032 "GET / HTTP/1.1" 200 13
ERROR 2024-03-27 23:10:39,126 log 11800 11032 Internal Server Error: /coin_values/
Traceback (most recent call last):
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\Seminars\myproject\myapp\views.py", line 23, in coin_values
    for text in value:
TypeError: 'NoneType' object is not iterable
ERROR 2024-03-27 23:10:39,126 basehttp 11800 11032 "GET /coin_values/ HTTP/1.1" 500 65208
INFO 2024-03-27 23:11:16,283 autoreload 11800 15576 D:\GeekBrains\Django\Seminars\myproject\myapp\views.py changed, reloading.
INFO 2024-03-27 23:11:16,700 autoreload 224 16032 Watching for file changes with StatReloader
INFO 2024-03-27 23:11:19,690 autoreload 15020 7960 Watching for file changes with StatReloader
INFO 2024-03-27 23:11:22,931 basehttp 15020 17696 "GET / HTTP/1.1" 200 13
ERROR 2024-03-27 23:11:26,430 log 15020 17696 Internal Server Error: /coin_values/
Traceback (most recent call last):
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\Seminars\myproject\myapp\views.py", line 23, in coin_values
    for text in value:
TypeError: 'NoneType' object is not iterable
ERROR 2024-03-27 23:11:26,430 basehttp 15020 17696 "GET /coin_values/ HTTP/1.1" 500 65224
INFO 2024-03-27 23:13:11,767 autoreload 15020 7960 D:\GeekBrains\Django\Seminars\myproject\myapp\views.py changed, reloading.
INFO 2024-03-27 23:13:12,262 autoreload 17956 4980 Watching for file changes with StatReloader
INFO 2024-03-27 23:13:13,603 basehttp 17956 17620 "GET / HTTP/1.1" 200 13
INFO 2024-03-27 23:13:16,690 basehttp 17956 17620 "GET /coin_values/ HTTP/1.1" 200 6
INFO 2024-03-27 23:13:17,508 basehttp 17956 17620 "GET /coin_values/ HTTP/1.1" 200 6
INFO 2024-03-27 23:13:41,016 autoreload 17956 4980 D:\GeekBrains\Django\Seminars\myproject\myapp\views.py changed, reloading.
INFO 2024-03-27 23:13:41,445 autoreload 16672 18024 Watching for file changes with StatReloader
INFO 2024-03-27 23:13:42,610 basehttp 16672 15296 "GET / HTTP/1.1" 200 13
INFO 2024-03-27 23:13:45,020 basehttp 16672 15296 "GET /coin_values/ HTTP/1.1" 200 6
INFO 2024-03-27 23:13:45,581 basehttp 16672 15296 "GET /coin_values/ HTTP/1.1" 200 6
INFO 2024-03-27 23:14:24,425 autoreload 16672 18024 D:\GeekBrains\Django\Seminars\myproject\myapp\views.py changed, reloading.
INFO 2024-03-27 23:14:24,840 autoreload 16324 10004 Watching for file changes with StatReloader
INFO 2024-03-27 23:14:28,239 basehttp 16324 7580 "GET / HTTP/1.1" 200 13
ERROR 2024-03-27 23:14:30,789 log 16324 7580 Internal Server Error: /coin_values/
Traceback (most recent call last):
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\Seminars\myproject\myapp\views.py", line 22, in coin_values
    print(type(list(value)))
               ^^^^^^^^^^^
TypeError: 'NoneType' object is not iterable
ERROR 2024-03-27 23:14:30,790 basehttp 16324 7580 "GET /coin_values/ HTTP/1.1" 500 65356
ERROR 2024-03-27 23:14:31,405 log 16324 7580 Internal Server Error: /coin_values/
Traceback (most recent call last):
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\Seminars\myproject\myapp\views.py", line 22, in coin_values
    print(type(list(value)))
               ^^^^^^^^^^^
TypeError: 'NoneType' object is not iterable
ERROR 2024-03-27 23:14:31,407 basehttp 16324 7580 "GET /coin_values/ HTTP/1.1" 500 65082
INFO 2024-03-27 23:15:18,038 autoreload 16324 10004 D:\GeekBrains\Django\Seminars\myproject\myapp\views.py changed, reloading.
INFO 2024-03-27 23:15:18,455 autoreload 17640 17124 Watching for file changes with StatReloader
INFO 2024-03-27 23:15:34,955 basehttp 17640 13060 "GET / HTTP/1.1" 200 13
INFO 2024-03-27 23:15:37,683 basehttp 17640 13060 "GET /coin_values/ HTTP/1.1" 200 6
INFO 2024-03-27 23:15:38,390 basehttp 17640 13060 "GET /coin_values/ HTTP/1.1" 200 6
INFO 2024-03-27 23:16:27,825 autoreload 17640 17124 D:\GeekBrains\Django\Seminars\myproject\myapp\views.py changed, reloading.
INFO 2024-03-27 23:16:28,223 autoreload 14560 15652 Watching for file changes with StatReloader
INFO 2024-03-27 23:16:29,708 basehttp 14560 10512 "GET / HTTP/1.1" 200 13
INFO 2024-03-27 23:16:32,572 basehttp 14560 10512 "GET /coin_values/ HTTP/1.1" 200 234
INFO 2024-03-27 23:16:33,237 basehttp 14560 10512 "GET /coin_values/ HTTP/1.1" 200 234
INFO 2024-03-27 23:16:52,916 autoreload 14560 15652 D:\GeekBrains\Django\Seminars\myproject\myapp\views.py changed, reloading.
INFO 2024-03-27 23:16:53,310 autoreload 12948 17832 Watching for file changes with StatReloader
INFO 2024-03-27 23:16:54,602 basehttp 12948 18192 "GET / HTTP/1.1" 200 13
INFO 2024-03-27 23:16:58,667 basehttp 12948 18192 "GET /coin_values/ HTTP/1.1" 200 240
INFO 2024-03-27 23:16:59,288 basehttp 12948 18192 "GET /coin_values/ HTTP/1.1" 200 240
INFO 2024-03-27 23:19:38,098 autoreload 12948 17832 D:\GeekBrains\Django\Seminars\myproject\myapp\models.py changed, reloading.
INFO 2024-03-27 23:19:38,516 autoreload 16252 8324 Watching for file changes with StatReloader
INFO 2024-03-27 23:20:07,169 autoreload 16252 8324 D:\GeekBrains\Django\Seminars\myproject\myapp\models.py changed, reloading.
INFO 2024-03-27 23:20:07,635 autoreload 6332 4992 Watching for file changes with StatReloader
INFO 2024-03-27 23:21:48,339 autoreload 6332 4992 D:\GeekBrains\Django\Seminars\myproject\myapp\models.py changed, reloading.
INFO 2024-03-27 23:21:48,788 autoreload 5732 5716 Watching for file changes with StatReloader
INFO 2024-03-27 23:23:10,977 autoreload 5732 5716 D:\GeekBrains\Django\Seminars\myproject\myapp\models.py changed, reloading.
INFO 2024-03-27 23:23:11,438 autoreload 10800 9376 Watching for file changes with StatReloader
INFO 2024-03-27 23:29:02,747 autoreload 10800 9376 D:\GeekBrains\Django\Seminars\myproject\myapp\models.py changed, reloading.
INFO 2024-03-27 23:29:03,180 autoreload 18248 11584 Watching for file changes with StatReloader
INFO 2024-03-27 23:45:52,471 autoreload 18416 7884 Watching for file changes with StatReloader
WARNING 2024-03-27 23:45:53,856 log 18416 5740 Not Found: /
WARNING 2024-03-27 23:45:53,856 basehttp 18416 5740 "GET / HTTP/1.1" 404 2271
INFO 2024-03-27 23:46:13,815 basehttp 18416 5740 "GET /blog HTTP/1.1" 200 7
INFO 2024-03-27 23:58:14,564 autoreload 13180 11312 Watching for file changes with StatReloader
WARNING 2024-03-27 23:58:15,715 log 13180 17680 Not Found: /
WARNING 2024-03-27 23:58:15,715 basehttp 13180 17680 "GET / HTTP/1.1" 404 2271
WARNING 2024-03-27 23:58:23,647 log 13180 17680 Not Found: /blog/fake_data
WARNING 2024-03-27 23:58:23,647 basehttp 13180 17680 "GET /blog/fake_data HTTP/1.1" 404 2570
INFO 2024-03-27 23:58:55,418 autoreload 13180 11312 D:\GeekBrains\Django\Seminars\myproject\blogapp\urls.py changed, reloading.
INFO 2024-03-27 23:58:55,822 autoreload 17404 10212 Watching for file changes with StatReloader
WARNING 2024-03-27 23:58:57,385 log 17404 9056 Not Found: /
WARNING 2024-03-27 23:58:57,387 basehttp 17404 9056 "GET / HTTP/1.1" 404 2271
WARNING 2024-03-27 23:59:06,265 log 17404 9056 Not Found: /blog/fake_data
WARNING 2024-03-27 23:59:06,266 basehttp 17404 9056 "GET /blog/fake_data HTTP/1.1" 404 2571
INFO 2024-03-28 00:01:21,275 autoreload 17404 10212 D:\GeekBrains\Django\Seminars\myproject\blogapp\views.py changed, reloading.
INFO 2024-03-28 00:01:21,710 autoreload 11056 16156 Watching for file changes with StatReloader
WARNING 2024-03-28 00:03:56,175 log 11056 13852 Not Found: /
WARNING 2024-03-28 00:03:56,175 basehttp 11056 13852 "GET / HTTP/1.1" 404 2271
INFO 2024-03-28 00:03:56,508 autoreload 11056 16156 D:\GeekBrains\Django\Seminars\myproject\blogapp\models.py changed, reloading.
INFO 2024-03-28 00:03:57,020 autoreload 1796 17996 Watching for file changes with StatReloader
WARNING 2024-03-28 00:04:07,494 log 1796 18400 Not Found: /blog/fake_data
WARNING 2024-03-28 00:04:07,494 basehttp 1796 18400 "GET /blog/fake_data HTTP/1.1" 404 2571
INFO 2024-03-28 00:05:53,040 autoreload 1796 17996 D:\GeekBrains\Django\Seminars\myproject\blogapp\urls.py changed, reloading.
INFO 2024-03-28 00:05:53,477 autoreload 14628 6052 Watching for file changes with StatReloader
INFO 2024-03-28 00:06:28,487 autoreload 16588 13716 Watching for file changes with StatReloader
INFO 2024-03-28 00:06:44,917 autoreload 16588 13716 D:\GeekBrains\Django\Seminars\myproject\blogapp\urls.py changed, reloading.
INFO 2024-03-28 00:06:47,720 autoreload 12532 15452 Watching for file changes with StatReloader
WARNING 2024-03-28 00:06:50,000 log 12532 16856 Not Found: /
WARNING 2024-03-28 00:06:50,002 basehttp 12532 16856 "GET / HTTP/1.1" 404 2271
WARNING 2024-03-28 00:06:59,928 log 12532 16856 Not Found: /blog/add_author
WARNING 2024-03-28 00:06:59,930 basehttp 12532 16856 "GET /blog/add_author HTTP/1.1" 404 2575
INFO 2024-03-28 00:08:39,702 autoreload 12532 15452 D:\GeekBrains\Django\Seminars\myproject\blogapp\urls.py changed, reloading.
INFO 2024-03-28 00:08:40,139 autoreload 17620 11980 Watching for file changes with StatReloader
INFO 2024-03-28 00:09:55,643 autoreload 10832 17428 Watching for file changes with StatReloader
WARNING 2024-03-28 00:11:00,982 log 10832 14792 Not Found: /
WARNING 2024-03-28 00:11:00,983 basehttp 10832 14792 "GET / HTTP/1.1" 404 2271
INFO 2024-03-28 00:11:05,021 basehttp 10832 14792 "GET /blog HTTP/1.1" 200 7
WARNING 2024-03-28 00:11:10,485 log 10832 14792 Not Found: /about/
WARNING 2024-03-28 00:11:10,486 basehttp 10832 14792 "GET /about/ HTTP/1.1" 404 2307
WARNING 2024-03-28 00:11:14,684 log 10832 14792 Not Found: /add_author
WARNING 2024-03-28 00:11:14,684 basehttp 10832 14792 "GET /add_author HTTP/1.1" 404 2319
WARNING 2024-03-28 00:11:29,562 log 10832 14792 Not Found: /blog/add_author
WARNING 2024-03-28 00:11:29,563 basehttp 10832 14792 "GET /blog/add_author HTTP/1.1" 404 2576
INFO 2024-03-28 00:14:21,863 autoreload 10832 17428 D:\GeekBrains\Django\Seminars\myproject\myproject\urls.py changed, reloading.
INFO 2024-03-28 00:14:22,274 autoreload 16984 14460 Watching for file changes with StatReloader
INFO 2024-03-28 00:14:24,573 autoreload 11688 8900 Watching for file changes with StatReloader
WARNING 2024-03-28 00:14:25,535 log 11688 5860 Not Found: /
WARNING 2024-03-28 00:14:25,535 basehttp 11688 5860 "GET / HTTP/1.1" 404 2510
INFO 2024-03-28 00:14:32,864 basehttp 11688 5860 "GET /index HTTP/1.1" 200 7
WARNING 2024-03-28 00:14:37,775 log 11688 5860 Not Found: /about/
WARNING 2024-03-28 00:14:37,775 basehttp 11688 5860 "GET /about/ HTTP/1.1" 404 2546
INFO 2024-03-28 00:14:41,548 basehttp 11688 5860 "GET /add_author HTTP/1.1" 301 0
ERROR 2024-03-28 00:14:41,621 log 11688 6356 Internal Server Error: /add_author/
Traceback (most recent call last):
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 329, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
sqlite3.IntegrityError: NOT NULL constraint failed: blogapp_author.birthday

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\Seminars\myproject\blogapp\views.py", line 13, in add_author
    author.save()
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\models\base.py", line 822, in save
    self.save_base(
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\models\base.py", line 909, in save_base
    updated = self._save_table(
              ^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\models\base.py", line 1067, in _save_table
    results = self._do_insert(
              ^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\models\base.py", line 1108, in _do_insert
    return manager._insert(
           ^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\models\query.py", line 1847, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\models\sql\compiler.py", line 1823, in execute_sql
    cursor.execute(sql, params)
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\backends\utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\backends\utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\backends\utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\backends\utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 329, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.IntegrityError: NOT NULL constraint failed: blogapp_author.birthday
ERROR 2024-03-28 00:14:41,626 basehttp 11688 6356 "GET /add_author/ HTTP/1.1" 500 148146
INFO 2024-03-28 00:15:43,084 autoreload 11688 8900 D:\GeekBrains\Django\Seminars\myproject\blogapp\views.py changed, reloading.
INFO 2024-03-28 00:15:43,494 autoreload 9932 5776 Watching for file changes with StatReloader
INFO 2024-03-28 00:15:55,913 autoreload 9932 5776 D:\GeekBrains\Django\Seminars\myproject\blogapp\urls.py changed, reloading.
INFO 2024-03-28 00:15:56,327 autoreload 2864 2620 Watching for file changes with StatReloader
INFO 2024-03-28 00:15:57,382 basehttp 2864 3448 "GET / HTTP/1.1" 200 7
WARNING 2024-03-28 00:16:00,472 log 2864 3448 Not Found: /about/
WARNING 2024-03-28 00:16:00,472 basehttp 2864 3448 "GET /about/ HTTP/1.1" 404 2541
INFO 2024-03-28 00:16:04,661 basehttp 2864 3448 "GET /add_author/ HTTP/1.1" 200 49
INFO 2024-03-28 00:18:34,894 autoreload 2864 2620 D:\GeekBrains\Django\Seminars\myproject\blogapp\models.py changed, reloading.
INFO 2024-03-28 00:19:32,053 autoreload 17480 8712 Watching for file changes with StatReloader
INFO 2024-03-28 00:19:33,064 basehttp 17480 17396 "GET / HTTP/1.1" 200 7
WARNING 2024-03-28 00:19:36,042 log 17480 17396 Not Found: /about/
WARNING 2024-03-28 00:19:36,043 basehttp 17480 17396 "GET /about/ HTTP/1.1" 404 2541
INFO 2024-03-28 00:19:37,496 basehttp 17480 17396 "GET /add_author/ HTTP/1.1" 200 49
INFO 2024-04-01 13:23:23,563 autoreload 6660 4796 Watching for file changes with StatReloader
INFO 2024-04-01 13:23:39,028 autoreload 6660 4796 D:\GeekBrains\Django\Seminars\myproject\myapp3\urls.py changed, reloading.
INFO 2024-04-01 13:23:39,425 autoreload 15932 6616 Watching for file changes with StatReloader
WARNING 2024-04-01 13:23:42,979 log 15932 2524 Not Found: /
WARNING 2024-04-01 13:23:42,979 basehttp 15932 2524 "GET / HTTP/1.1" 404 2384
WARNING 2024-04-01 13:23:48,386 log 15932 2524 Not Found: /myapp3
WARNING 2024-04-01 13:23:48,386 basehttp 15932 2524 "GET /myapp3 HTTP/1.1" 404 2420
WARNING 2024-04-01 13:23:51,418 log 15932 2524 Not Found: /myapp3/index
WARNING 2024-04-01 13:23:51,418 basehttp 15932 2524 "GET /myapp3/index HTTP/1.1" 404 2715
INFO 2024-04-01 13:24:10,161 basehttp 15932 2524 "GET /myapp3/index/index HTTP/1.1" 301 0
ERROR 2024-04-01 13:24:10,264 log 15932 9524 Internal Server Error: /myapp3/index/index/
Traceback (most recent call last):
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\Seminars\myproject\myapp3\views.py", line 6, in index
    return render('myapp3/index.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\backends\django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\loaders\base.py", line 17, in get_template
    for origin in self.get_template_sources(template_name):
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\loaders\cached.py", line 70, in get_template_sources
    yield from loader.get_template_sources(template_name)
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\loaders\filesystem.py", line 35, in get_template_sources
    name = safe_join(template_dir, template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\utils\_os.py", line 17, in safe_join
    final_path = abspath(join(base, *paths))
                         ^^^^^^^^^^^^^^^^^^
  File "<frozen ntpath>", line 147, in join
  File "<frozen genericpath>", line 152, in _check_arg_types
TypeError: join() argument must be str, bytes, or os.PathLike object, not 'dict'
ERROR 2024-04-01 13:24:10,268 basehttp 15932 9524 "GET /myapp3/index/index/ HTTP/1.1" 500 119907
INFO 2024-04-01 13:25:02,303 autoreload 15932 6616 D:\GeekBrains\Django\Seminars\myproject\myapp3\urls.py changed, reloading.
INFO 2024-04-01 13:25:02,716 autoreload 5588 17336 Watching for file changes with StatReloader
WARNING 2024-04-01 13:25:03,963 log 5588 18144 Not Found: /
WARNING 2024-04-01 13:25:03,964 basehttp 5588 18144 "GET / HTTP/1.1" 404 2384
WARNING 2024-04-01 13:25:11,875 log 5588 18144 Not Found: /index/
WARNING 2024-04-01 13:25:11,875 basehttp 5588 18144 "GET /index/ HTTP/1.1" 404 2420
WARNING 2024-04-01 13:25:12,911 log 5588 18144 Not Found: /index/
WARNING 2024-04-01 13:25:12,911 basehttp 5588 18144 "GET /index/ HTTP/1.1" 404 2420
WARNING 2024-04-01 13:25:24,657 log 5588 18144 Not Found: /myapp3
WARNING 2024-04-01 13:25:24,658 basehttp 5588 18144 "GET /myapp3 HTTP/1.1" 404 2420
INFO 2024-04-01 13:25:26,546 basehttp 5588 18144 "GET /myapp3/index HTTP/1.1" 301 0
ERROR 2024-04-01 13:25:26,577 log 5588 1768 Internal Server Error: /myapp3/index/
Traceback (most recent call last):
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\Seminars\myproject\myapp3\views.py", line 6, in index
    return render('myapp3/index.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\backends\django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\loaders\base.py", line 17, in get_template
    for origin in self.get_template_sources(template_name):
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\loaders\cached.py", line 70, in get_template_sources
    yield from loader.get_template_sources(template_name)
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\loaders\filesystem.py", line 35, in get_template_sources
    name = safe_join(template_dir, template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\utils\_os.py", line 17, in safe_join
    final_path = abspath(join(base, *paths))
                         ^^^^^^^^^^^^^^^^^^
  File "<frozen ntpath>", line 147, in join
  File "<frozen genericpath>", line 152, in _check_arg_types
TypeError: join() argument must be str, bytes, or os.PathLike object, not 'dict'
ERROR 2024-04-01 13:25:26,580 basehttp 5588 1768 "GET /myapp3/index/ HTTP/1.1" 500 120108
INFO 2024-04-01 13:26:13,304 autoreload 5588 17336 D:\GeekBrains\Django\Seminars\myproject\myapp3\views.py changed, reloading.
INFO 2024-04-01 13:26:13,716 autoreload 11744 4836 Watching for file changes with StatReloader
INFO 2024-04-01 13:26:14,735 basehttp 11744 1712 "GET /myapp3/index/ HTTP/1.1" 200 1266
INFO 2024-04-01 13:28:43,356 autoreload 11744 4836 D:\GeekBrains\Django\Seminars\myproject\myapp3\urls.py changed, reloading.
INFO 2024-04-01 13:28:43,773 autoreload 6692 4012 Watching for file changes with StatReloader
WARNING 2024-04-01 13:28:44,751 log 6692 16780 Not Found: /
WARNING 2024-04-01 13:28:44,751 basehttp 6692 16780 "GET / HTTP/1.1" 404 2384
INFO 2024-04-01 13:28:49,209 basehttp 6692 16780 "GET /myapp3 HTTP/1.1" 301 0
INFO 2024-04-01 13:28:49,214 basehttp 6692 4648 "GET /myapp3/ HTTP/1.1" 200 1075
INFO 2024-04-01 13:29:11,871 basehttp 6692 4648 "GET /myapp3/ HTTP/1.1" 200 1075
INFO 2024-04-01 13:29:12,562 autoreload 6692 4012 D:\GeekBrains\Django\Seminars\myproject\myapp3\views.py changed, reloading.
INFO 2024-04-01 13:29:12,983 autoreload 3708 16740 Watching for file changes with StatReloader
INFO 2024-04-01 13:29:13,129 basehttp 3708 17252 "GET /myapp3/ HTTP/1.1" 200 859
INFO 2024-04-01 13:29:14,717 basehttp 3708 17252 "GET /myapp3/index/ HTTP/1.1" 200 1270
INFO 2024-04-01 13:29:16,665 basehttp 3708 17252 "GET /myapp3/about HTTP/1.1" 301 0
INFO 2024-04-01 13:29:16,668 basehttp 3708 15204 "GET /myapp3/about/ HTTP/1.1" 200 1074
INFO 2024-04-01 13:29:18,168 basehttp 3708 15204 "GET /myapp3/index/ HTTP/1.1" 200 1270
INFO 2024-04-01 13:29:39,707 basehttp 3708 15204 "GET /myapp3/ HTTP/1.1" 200 859
INFO 2024-04-01 13:29:40,663 autoreload 3708 16740 D:\GeekBrains\Django\Seminars\myproject\myapp3\views.py changed, reloading.
INFO 2024-04-01 13:29:41,087 autoreload 11456 9296 Watching for file changes with StatReloader
INFO 2024-04-01 13:29:41,291 basehttp 11456 10864 "GET /myapp3/ HTTP/1.1" 200 859
INFO 2024-04-01 13:29:42,738 basehttp 11456 10864 "GET /myapp3/index/ HTTP/1.1" 200 1261
INFO 2024-04-01 13:29:46,345 basehttp 11456 10864 "GET /myapp3/about/ HTTP/1.1" 200 1074
INFO 2024-04-01 13:29:47,664 basehttp 11456 10864 "GET /myapp3/index/ HTTP/1.1" 200 1261
INFO 2024-04-01 13:29:49,255 basehttp 11456 10864 "GET /myapp3/about/ HTTP/1.1" 200 1074
WARNING 2024-04-01 13:40:17,996 log 11456 7004 Not Found: /
WARNING 2024-04-01 13:40:17,996 basehttp 11456 7004 "GET / HTTP/1.1" 404 2384
ERROR 2024-04-01 13:40:24,135 log 11456 7004 Internal Server Error: /myapp3/
Traceback (most recent call last):
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\base.py", line 502, in parse
    compile_func = self.tags[command]
                   ~~~~~~~~~^^^^^^^^^
KeyError: 'end'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\Seminars\myproject\myapp3\views.py", line 16, in main
    return render(request, 'myapp3/main.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\backends\django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    return parser.parse()
           ^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\base.py", line 510, in parse
    raise self.error(token, e)
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\base.py", line 508, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\loader_tags.py", line 295, in do_extends
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\base.py", line 510, in parse
    raise self.error(token, e)
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\base.py", line 508, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\loader_tags.py", line 234, in do_block
    nodelist = parser.parse(("endblock",))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\base.py", line 504, in parse
    self.invalid_block_tag(token, command, parse_until)
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\template\base.py", line 555, in invalid_block_tag
    raise self.error(
django.template.exceptions.TemplateSyntaxError: Invalid block tag on line 9: 'end', expected 'endblock'. Did you forget to register or load this tag?
ERROR 2024-04-01 13:40:24,139 basehttp 11456 7004 "GET /myapp3/ HTTP/1.1" 500 158859
INFO 2024-04-01 13:43:37,180 basehttp 11456 7004 "GET /myapp3/ HTTP/1.1" 200 865
INFO 2024-04-01 13:43:39,675 basehttp 11456 7004 "GET /myapp3/index/ HTTP/1.1" 200 1262
INFO 2024-04-01 13:43:41,797 basehttp 11456 7004 "GET /myapp3/about/ HTTP/1.1" 200 1071
INFO 2024-04-01 14:51:58,851 basehttp 11456 11900 "GET /myapp3/index/ HTTP/1.1" 200 1262
INFO 2024-04-01 14:52:01,842 basehttp 11456 11900 "GET /myapp3/about/ HTTP/1.1" 200 1071
INFO 2024-04-01 14:52:04,153 basehttp 11456 11900 "GET /myapp3/index/ HTTP/1.1" 200 1262
INFO 2024-04-01 15:01:03,159 autoreload 11456 9296 D:\GeekBrains\Django\Seminars\myproject\myapp3\models.py changed, reloading.
INFO 2024-04-01 15:01:03,583 autoreload 14900 4484 Watching for file changes with StatReloader
INFO 2024-04-01 15:07:07,209 autoreload 14900 4484 D:\GeekBrains\Django\Seminars\myproject\myapp3\models.py changed, reloading.
INFO 2024-04-01 15:07:07,699 autoreload 6188 2004 Watching for file changes with StatReloader
INFO 2024-04-01 15:45:09,600 autoreload 11576 7728 Watching for file changes with StatReloader
WARNING 2024-04-01 15:45:10,880 log 11576 1248 Not Found: /
WARNING 2024-04-01 15:45:10,881 basehttp 11576 1248 "GET / HTTP/1.1" 404 2384
WARNING 2024-04-01 15:45:15,827 log 11576 1248 Not Found: /myapp3/
WARNING 2024-04-01 15:45:15,828 basehttp 11576 1248 "GET /myapp3/ HTTP/1.1" 404 3024
INFO 2024-04-01 15:45:25,937 basehttp 11576 1248 "GET /myapp3/cub/10 HTTP/1.1" 301 0
INFO 2024-04-01 15:45:25,946 basehttp 11576 10212 "GET /myapp3/cub/10/ HTTP/1.1" 200 1234
WARNING 2024-04-01 15:48:38,591 log 11576 10212 Not Found: /
WARNING 2024-04-01 15:48:38,591 basehttp 11576 10212 "GET / HTTP/1.1" 404 2384
WARNING 2024-04-01 15:48:42,296 log 11576 10212 Not Found: /myapp3/
WARNING 2024-04-01 15:48:42,296 basehttp 11576 10212 "GET /myapp3/ HTTP/1.1" 404 3024
WARNING 2024-04-01 15:48:43,490 log 11576 10212 Not Found: /myapp3/
WARNING 2024-04-01 15:48:43,491 basehttp 11576 10212 "GET /myapp3/ HTTP/1.1" 404 3024
WARNING 2024-04-01 15:48:51,862 log 11576 10212 Not Found: /
WARNING 2024-04-01 15:48:51,862 basehttp 11576 10212 "GET / HTTP/1.1" 404 2384
WARNING 2024-04-01 15:48:54,848 log 11576 10212 Not Found: /coin_values/
WARNING 2024-04-01 15:48:54,848 basehttp 11576 10212 "GET /coin_values/ HTTP/1.1" 404 2438
WARNING 2024-04-01 15:48:55,879 log 11576 10212 Not Found: /cub
WARNING 2024-04-01 15:48:55,879 basehttp 11576 10212 "GET /cub HTTP/1.1" 404 2411
WARNING 2024-04-01 15:48:57,895 log 11576 10212 Not Found: /cub/10
WARNING 2024-04-01 15:48:57,895 basehttp 11576 10212 "GET /cub/10 HTTP/1.1" 404 2420
WARNING 2024-04-01 15:49:08,787 log 11576 10212 Not Found: /myapp/cub/10
WARNING 2024-04-01 15:49:08,787 basehttp 11576 10212 "GET /myapp/cub/10 HTTP/1.1" 404 2438
INFO 2024-04-01 15:49:27,667 basehttp 11576 10212 "GET /myapp3/cub/10/ HTTP/1.1" 200 1322
INFO 2024-04-01 15:52:53,427 autoreload 11576 7728 D:\GeekBrains\Django\Seminars\myproject\myapp3\urls.py changed, reloading.
INFO 2024-04-01 15:52:53,958 autoreload 9096 7024 Watching for file changes with StatReloader
WARNING 2024-04-01 15:52:55,181 log 9096 3592 Not Found: /
WARNING 2024-04-01 15:52:55,182 basehttp 9096 3592 "GET / HTTP/1.1" 404 2384
WARNING 2024-04-01 15:52:58,341 log 9096 3592 Not Found: /myapp3/
WARNING 2024-04-01 15:52:58,341 basehttp 9096 3592 "GET /myapp3/ HTTP/1.1" 404 3410
INFO 2024-04-01 15:53:01,865 basehttp 9096 3592 "GET /myapp3/cub/10/ HTTP/1.1" 200 892
INFO 2024-04-01 15:53:52,617 basehttp 9096 3592 "GET /myapp3/numbers/10 HTTP/1.1" 301 0
INFO 2024-04-01 15:53:52,620 basehttp 9096 18096 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1327
INFO 2024-04-01 15:54:01,008 basehttp 9096 18096 "GET /myapp3/coin/10 HTTP/1.1" 301 0
ERROR 2024-04-01 15:54:01,035 log 9096 10444 Internal Server Error: /myapp3/coin/10/
Traceback (most recent call last):
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\Seminars\myproject\myapp3\views.py", line 48, in coin
    my_dict[choice['Орёл', 'Решка']] = datetime.datetime.now().time()
            ~~~~~~^^^^^^^^^^^^^^^^^
TypeError: 'method' object is not subscriptable
ERROR 2024-04-01 15:54:01,037 basehttp 9096 10444 "GET /myapp3/coin/10/ HTTP/1.1" 500 65446
INFO 2024-04-01 15:54:47,252 autoreload 9096 7024 D:\GeekBrains\Django\Seminars\myproject\myapp3\views.py changed, reloading.
INFO 2024-04-01 15:54:47,698 autoreload 3228 10160 Watching for file changes with StatReloader
ERROR 2024-04-01 15:54:49,338 log 3228 2004 Internal Server Error: /myapp3/coin/10/
Traceback (most recent call last):
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GeekBrains\Django\Seminars\myproject\myapp3\views.py", line 48, in coin
    my_dict[choice('Орёл', 'Решка')] = datetime.datetime.now().time()
            ^^^^^^^^^^^^^^^^^^^^^^^
TypeError: Random.choice() takes 2 positional arguments but 3 were given
ERROR 2024-04-01 15:54:49,339 basehttp 3228 2004 "GET /myapp3/coin/10/ HTTP/1.1" 500 65643
INFO 2024-04-01 15:55:53,128 autoreload 3228 10160 D:\GeekBrains\Django\Seminars\myproject\myapp3\views.py changed, reloading.
INFO 2024-04-01 15:55:53,605 autoreload 8148 13576 Watching for file changes with StatReloader
INFO 2024-04-01 15:56:03,673 basehttp 8148 16740 "GET /myapp3/coin/10/ HTTP/1.1" 200 850
INFO 2024-04-01 15:56:11,403 basehttp 8148 16740 "GET /myapp3/coin/100/ HTTP/1.1" 200 850
INFO 2024-04-01 16:06:09,246 autoreload 8148 13576 D:\GeekBrains\Django\Seminars\myproject\myapp3\views.py changed, reloading.
INFO 2024-04-01 16:06:09,693 autoreload 15368 17044 Watching for file changes with StatReloader
INFO 2024-04-01 16:06:11,971 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 4720
INFO 2024-04-01 16:06:55,776 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 4728
INFO 2024-04-01 16:07:01,800 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 4708
INFO 2024-04-01 16:07:02,818 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 4710
INFO 2024-04-01 16:07:03,484 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 4692
INFO 2024-04-01 16:07:04,084 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 4704
INFO 2024-04-01 16:07:04,670 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 4706
INFO 2024-04-01 16:07:05,202 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 4704
INFO 2024-04-01 16:07:05,629 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 4722
INFO 2024-04-01 16:07:07,024 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 4732
INFO 2024-04-01 16:07:25,991 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 4718
INFO 2024-04-01 16:07:26,721 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 5614
INFO 2024-04-01 16:07:27,402 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 5614
INFO 2024-04-01 16:07:27,904 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 5612
INFO 2024-04-01 16:07:28,219 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 5626
INFO 2024-04-01 16:07:28,408 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 5618
INFO 2024-04-01 16:07:28,565 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 5622
INFO 2024-04-01 16:07:29,037 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 5610
INFO 2024-04-01 16:07:29,428 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 5622
INFO 2024-04-01 16:07:29,794 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 5618
INFO 2024-04-01 16:07:30,124 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 5620
INFO 2024-04-01 16:07:30,311 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 5612
INFO 2024-04-01 16:07:30,477 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 5606
INFO 2024-04-01 16:07:30,634 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 5600
INFO 2024-04-01 16:07:30,792 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 5620
INFO 2024-04-01 16:07:30,941 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 5616
INFO 2024-04-01 16:07:31,100 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 5614
INFO 2024-04-01 16:07:31,258 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 5602
INFO 2024-04-01 16:07:31,415 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 5636
INFO 2024-04-01 16:07:31,565 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 5620
INFO 2024-04-01 16:07:31,715 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 5608
INFO 2024-04-01 16:07:45,957 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 5610
INFO 2024-04-01 16:07:46,618 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 5606
INFO 2024-04-01 16:08:01,527 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 5614
INFO 2024-04-01 16:08:02,120 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 7138
INFO 2024-04-01 16:08:02,532 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 7126
INFO 2024-04-01 16:08:02,704 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 7122
INFO 2024-04-01 16:08:02,855 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 7104
INFO 2024-04-01 16:08:03,012 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 7116
INFO 2024-04-01 16:08:03,177 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 7120
INFO 2024-04-01 16:08:03,382 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 7118
INFO 2024-04-01 16:08:03,530 basehttp 15368 17864 "GET /myapp3/coin/100/ HTTP/1.1" 200 7136
INFO 2024-04-01 16:09:44,423 autoreload 15368 17044 D:\GeekBrains\Django\Seminars\myproject\myapp3\views.py changed, reloading.
INFO 2024-04-01 16:09:44,864 autoreload 5820 2436 Watching for file changes with StatReloader
INFO 2024-04-01 16:09:51,285 basehttp 5820 8900 "GET /myapp3/cub/10/ HTTP/1.1" 200 1125
INFO 2024-04-01 16:10:06,629 basehttp 5820 8900 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1147
INFO 2024-04-01 16:11:15,936 basehttp 5820 8900 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1219
INFO 2024-04-01 16:11:16,633 basehttp 5820 8900 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1219
INFO 2024-04-01 16:11:17,570 basehttp 5820 8900 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1220
INFO 2024-04-01 16:11:31,052 autoreload 17900 4168 Watching for file changes with StatReloader
INFO 2024-04-01 16:11:33,433 basehttp 17900 17284 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1218
INFO 2024-04-01 16:11:34,139 basehttp 17900 17284 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1218
INFO 2024-04-01 16:12:00,357 autoreload 16852 8880 Watching for file changes with StatReloader
INFO 2024-04-01 16:12:03,472 basehttp 16852 16108 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1261
INFO 2024-04-01 16:12:04,399 basehttp 16852 16108 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1259
INFO 2024-04-01 16:12:05,599 basehttp 16852 16108 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1261
INFO 2024-04-01 16:12:06,027 basehttp 16852 16108 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1260
INFO 2024-04-01 16:12:06,192 basehttp 16852 16108 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1261
INFO 2024-04-01 16:12:06,364 basehttp 16852 16108 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1260
INFO 2024-04-01 16:12:06,537 basehttp 16852 16108 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1260
INFO 2024-04-01 16:12:06,875 basehttp 16852 16108 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1259
INFO 2024-04-01 16:12:07,204 basehttp 16852 16108 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1261
INFO 2024-04-01 16:12:07,601 basehttp 16852 16108 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1259
INFO 2024-04-01 16:12:07,978 basehttp 16852 16108 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1260
INFO 2024-04-01 16:12:08,382 basehttp 16852 16108 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1258
INFO 2024-04-01 16:12:09,072 basehttp 16852 16108 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1261
INFO 2024-04-01 16:12:09,319 basehttp 16852 16108 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1258
INFO 2024-04-01 16:12:09,491 basehttp 16852 16108 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1260
INFO 2024-04-01 16:12:09,642 basehttp 16852 16108 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1260
INFO 2024-04-01 16:12:09,791 basehttp 16852 16108 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1258
INFO 2024-04-01 16:12:10,055 basehttp 16852 16108 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1261
INFO 2024-04-01 16:12:10,264 basehttp 16852 16108 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1260
INFO 2024-04-01 16:12:10,430 basehttp 16852 16108 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1260
INFO 2024-04-01 16:12:10,564 basehttp 16852 16108 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1260
INFO 2024-04-01 16:12:10,714 basehttp 16852 16108 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1259
INFO 2024-04-01 16:12:10,864 basehttp 16852 16108 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1259
INFO 2024-04-01 16:12:11,021 basehttp 16852 16108 "GET /myapp3/numbers/10/ HTTP/1.1" 200 1261
